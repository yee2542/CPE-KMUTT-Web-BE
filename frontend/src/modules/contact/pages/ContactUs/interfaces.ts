import { GetContactQuery } from 'common/generated/generated-types';

export interface ContactUsProps {
  data?: GetContactQuery;
}
