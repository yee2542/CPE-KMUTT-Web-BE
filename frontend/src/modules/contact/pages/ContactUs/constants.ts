import { CrumbProps } from 'common/components/Layouts/CommonLayout/types';

import { CONTACT_US_PAGE_CRUMB } from 'modules/contact/constants';
import { HOME_PAGE_CRUMB } from 'modules/home/constants';

export const HEADER_BACKGROUND_IMAGE = '/images/contact_us_bg_header.png';

export const BREADCRUMB: CrumbProps[] = [HOME_PAGE_CRUMB, CONTACT_US_PAGE_CRUMB];
