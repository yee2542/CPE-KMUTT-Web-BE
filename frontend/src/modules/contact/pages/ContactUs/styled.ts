import { Row } from 'antd';
import styled from 'styled-components';

import colors from 'common/constants/colors';

const SPACE_FOR_CONTACT_DETAIL = '50px';

export const StyledRow = styled(Row)`
  width: 100%;
`;

export const Address = styled.div`
  color: ${colors.GRAY_1};
  margin-bottom: ${SPACE_FOR_CONTACT_DETAIL};
`;

export const ContactDetail = styled.div`
  color: ${colors.GRAY_2};
  font-weight: medium;
`;
