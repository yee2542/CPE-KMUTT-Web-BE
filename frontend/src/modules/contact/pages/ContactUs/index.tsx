import { Col } from 'antd';
import Head from 'next/head';

import Container from 'common/components/Container';
import Heading from 'common/components/Heading';
import CommonLayout from 'common/components/Layouts/CommonLayout';
import colors from 'common/constants/colors';

import BadgeContactSection from 'modules/contact/components/BadgeContactSection';
import {
  CONTACT_US_TITLE,
  DEFAULT_CONTACT_ADDRESS,
  DEFAULT_CONTACT_EMAIL,
  DEFAULT_CONTACT_PHONE,
  DEFAULT_GMAPS_URL,
} from 'modules/contact/constants';
import { ContactUsProps } from 'modules/contact/pages/ContactUs/interfaces';

import { BREADCRUMB, HEADER_BACKGROUND_IMAGE } from './constants';
import { Address, ContactDetail, ContactSection, StyledRow } from './styled';

const ContactUs: React.FC<ContactUsProps> = ({ data }) => {
  const address = data?.contact?.address_content || DEFAULT_CONTACT_ADDRESS;
  const phone = data?.contact?.contact_phone || DEFAULT_CONTACT_PHONE;
  const email = data?.contact?.contact_email || DEFAULT_CONTACT_EMAIL;
  const mapUrl = data?.contact?.google_map_link || DEFAULT_GMAPS_URL;

  return (
    <>
      <Head>
        <title>{CONTACT_US_TITLE}</title>
      </Head>
      <CommonLayout
        header="ติดต่อเรา"
        navigate={BREADCRUMB}
        headerBackgroundImage={HEADER_BACKGROUND_IMAGE}>
        <Container>
          <ContactSection>
            <StyledRow>
              <Col span={12}>
                <StyledRow>
                  <Heading variant="h2" color={colors.PRIMARY_COLOR}>
                    ที่อยู่
                  </Heading>
                  <Address>{address}</Address>
                </StyledRow>
                <StyledRow>
                  <Heading variant="h2" color={colors.PRIMARY_COLOR}>
                    ติดต่อ
                  </Heading>
                  <ContactDetail>
                    <p>{phone}</p>
                    <p>{email}</p>
                  </ContactDetail>
                </StyledRow>
              </Col>
              <Col span={12}>
                <iframe
                  title="KMUTT CPE"
                  src={mapUrl}
                  width="100%"
                  height="300"
                  style={{ border: 0 }}
                  loading="lazy"></iframe>
              </Col>
            </StyledRow>
          </ContactSection>
        </Container>
        <BadgeContactSection />
      </CommonLayout>
    </>
  );
};

export default ContactUs;
