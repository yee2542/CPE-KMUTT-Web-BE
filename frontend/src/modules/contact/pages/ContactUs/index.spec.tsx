import * as React from 'react';

import { shallow } from 'enzyme';
import ContactUs from 'pages/contact-us';

describe('Contact Us Page', () => {
  it('Contact Us to match snapshots', () => {
    const wrapper = shallow(<ContactUs />);
    expect(wrapper).toMatchSnapshot();
  });
});
