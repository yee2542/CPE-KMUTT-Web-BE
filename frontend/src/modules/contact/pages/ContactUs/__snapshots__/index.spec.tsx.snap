// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Us Page Contact Us to match snapshot 1`] = `
<Fragment>
  <Head>
    <title>
      ติดต่อเรา
    </title>
  </Head>
  <CommonLayout
    header="ติดต่อเรา"
    headerBackgroundImage="/images/contact_us_bg_header.png"
    navigate={
      Array [
        Object {
          "link": "/",
          "title": "หนัาหลัก",
        },
        Object {
          "link": "/contact-us",
          "title": "ติดต่อเรา",
        },
      ]
    }
  >
    <Container>
      <styled__ContactSection>
        <styled__StyledRow>
          <Col
            span={12}
          >
            <styled__StyledRow>
              <Heading
                color="#4A60AC"
                variant="h2"
              >
                ที่อยู่
              </Heading>
              <styled__Address>
                
ภาควิชาวิศวกรรมคอมพิวเตอร์ คณะวิศวกรรมศาสตร์ 
ชั้น 10-11, อาคารวิศววัฒนะ,
มหาวิทยาลัยเทคโนโลยีพระจอมเกล้าธนบุรี 126 ถนนประชาอุทิศ แขวงบางมด
เขตทุ่งครุ กรุงเทพฯ 10140

              </styled__Address>
            </styled__StyledRow>
            <styled__StyledRow>
              <Heading
                color="#4A60AC"
                variant="h2"
              >
                ติดต่อ
              </Heading>
              <styled__ContactDetail>
                <p>
                  (+66) 0 2470 9388
                </p>
                <p>
                  nongyao.jam@mail.kmutt.ac.th
                </p>
              </styled__ContactDetail>
            </styled__StyledRow>
          </Col>
          <Col
            span={12}
          >
            <iframe
              height="300"
              loading="lazy"
              src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d1630.1330306733955!2d100.49378030791785!3d13.649757914204436!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x30e2a252224fdc01%3A0xdb5f69b03ef7ef07!2sComputer%20Engineering%20Department%2C%20KMUTT!5e0!3m2!1sen!2sth!4v1649610824894!5m2!1sen!2sth"
              style={
                Object {
                  "border": 0,
                }
              }
              title="KMUTT CPE"
              width="100%"
            />
          </Col>
        </styled__StyledRow>
      </styled__ContactSection>
    </Container>
    <BadgeContactSection />
  </CommonLayout>
</Fragment>
`;
